daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -314,7 +316,13 @@ object Speedy {
           case SpeedyHungry(res: SResult) => result = res //stop
           case serr: SError =>
             serr match {
-              case _: SErrorDamlException if tryHandleException => () // outer loop will run again
+              case e: SErrorDamlException if tryHandleException => // outer loop will run again
+                e match {
+                    case DamlEUserError(message) =>
+                      userErrorMessage = Some(message)
```

Comment:
it looks like you never reset this which I believe can cause problems in the following scenario:
1. First, you call `submitMustFailMsg` on something that crashes with `abort`.
2. Next, you call `submitMustFailMsg` on something that crashes due to something else, e.g., missing authorization. That will neither change or reset the message so you will return the message set in 1.